.github/workflows/vfd-main.yml

```diff
@@ -0,0 +1,66 @@
+name: hdf5 VFD CI main
+
+on:
+  workflow_call:
+    inputs:
+      build_sys:
+        description: "build system used"
+        required: true
+        type: string
+      build_mode:
+        description: "release vs. debug build"
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  # Test HDF5 VFDs that are always built by default
+  #hdf5_vfd_standard:
```

Comment:
These can be filled in afterward, but I listed the ones I know about here. It would be nice for all of the "standard" VFDs to all use the same build of HDF5 instead of all being a separate build, but it would also be nice to be able to test the different VFDs separately to better parallelize.